zanata-war/src/main/java/org/zanata/action/PasswordResetAction.java

```diff
@@ -108,9 +110,9 @@ public void validateActivationKey() {
 
     @Transactional
     public String changePassword() {
-        // need to get username from DAO due to lazy loading of account in key
         String username =
-            accountResetPasswordKeyDAO.getUsername(key.getKeyHash());
```

Comment:
@broccolinisoup This is the query I was referring to. If you follow the `getUsername` method on this DAO you'll see the query.